slatedb/src/db.rs

```diff
@@ -531,6 +531,61 @@ impl DbInner {
         Ok(())
     }
 
+    async fn preload_meta_cache(&self) -> Result<(), SlateDBError> {
+        let Some(preload_level) = self.settings.meta_cache_preload else {
+            return Ok(());
+        };
+
+        if !self.table_store.has_cache() {
+            warn!("Meta cache preload was requested, but no cache is configured; skipping");
+            return Ok(());
+        }
+
+        let current_state = self.state.read().state();
+        let max_parallelism = compute_max_parallel(
+            current_state.manifest.core.l0.len(),
+            &current_state.manifest.core.compacted,
+            4,
+        );
+
+        // Load compacted SSTs first so that L0 entries are the most recently added
+        // to the cache, keeping them hotter when capacity is tight.
+        let mut handles_to_load: Vec<SsTableHandle> = Vec::new();
+        if matches!(preload_level, PreloadLevel::AllSst) {
+            handles_to_load.extend(
+                current_state
+                    .manifest
+                    .core
+                    .compacted
+                    .iter()
+                    .flat_map(|level| level.ssts.iter().cloned()),
+            );
+        }
+        handles_to_load.extend(current_state.manifest.core.l0.iter().cloned());
+
+        if handles_to_load.is_empty() {
+            return Ok(());
+        }
+
+        let table_store = self.table_store.clone();
+
+        let _results = build_concurrent(handles_to_load, max_parallelism, move |handle| {
```

Comment:
Instead of discarding `_result`, mind doing an `.await?`? I think we log in the builder when it returns an Err, so the builder should treat the error as non-fatal, which is correct (after logging a `warn`).